src/python/README.md

```diff
@@ -0,0 +1,217 @@
+6. `extractor`: bytes required of key genertion. Automatically generated by the constructor. *This value is not specified by the user.*
+
+*Note that once these parameters are set, they will not be altered again for that code run, meaning they will be globally constant throughout the code.*
+
+#### FuzzyState (Class):
+
+The state is retained for recovering keys, this is what is sometimes refered to as the "payload". It's created at the time of the call to `GenerateSecret()`. It's up to the caller to save this information in the form of a JSON file. This is easy to do since the string represenation of this object, as returned by `__repr__()`, is a string containing the JSON representation. This information is required by `RecoverSecret()`.
+
+#### GenerateSecret (Function):
+`GenerateSecret()` establishes the original words of the vault. In addition, it returns a state and a list of 512-bit keys. The state contains the information needed to recover the keys with `RecoverSecret()`.
+
+*Return Output:*
+1. `FuzzyState` object.
+2. A list of `key_count` 512-bit keys.
+
+*Input:*
+1. `params`: This is an object of the class `InputParams` that contains the necessary input.
+2. `original_words`: A list of unique of integers in the range 0 .. (`corpus_size` - 1) representing the secret words needed to recover the keys.
+3. `key_count`: The initial number of keys requested.
+
+#### RecoverSecret (Function):
+`RecoverSecret()` recovers a `key_count` list of keys. 
+
+*Return Output:*
+
+1. A list of keys if successful. Otherwise, it throws a `FuzzyError exception` if the keys cannot be recovered.
+
+*Input:*
+1. `state`: The FuzzyState created by GenerateSecret.
+2. `recovery_words`: A guess of the original words. These are unique integers in the range 0 .. `corpus_size` - 1.
+3. `key_count`: The number of keys to be generated.
+
+---
+### Tests and Demos 
+<br/>
+After installing the necessary prerequisites and getting familiar with the code, we've provided a tutorial you can use to test and demo our scheme...
+
+First, you will need to need to change your directory to the right PATH where the python files are located, as such:
+
+```shell
+$ cd PATH/
+```
+After that, you can start your demo by setting the parameters for generating your cryptographic key:
+
+```shell
+$ python3 gen_params.py
+``` 
+You can also add an option to set the path to another JSON file to hold the generated input parameters (default= `params.json`):
+
+ ```shell
+$ python3 gen_params.py --params-path TEXT
+``` 
+Specify your parameters (for the purpose of this tutorial, we will use the following example):
+```
+set-size: 9
+correct-threshold: 7
+corpus-size: 7000
+writing parameters to params.json
+``` 
+If you'd like to review the parameters you've set, you can simply open `params.json` (by default):
+```shell
+$ cat params.json
+``` 
+Output for our example:
+```
+{
+  "setsize": 9,
```

Comment:
Minor nit but I see you are mixing the JSON styling here, suggest using something like camelCasing or snake_casing consistently

Reply:
Good point - perhaps we can link to the style guide we chose for Sidetree? (you happen to have a link handy?)